### PR TITLE
Prep for TinyGo's v0.28 encoding/json Support

### PR DESCRIPTION
**Pull request by syke99**
This PR removes the fastjson dependency in favor of using the stdlib's encoding/json package as it will be fully supported by TinyGo's v0.28 release

### CODE REVIEW

**Comment by nilslice** [comment]
thanks @syke99! we just updated our CI for this repo to use the latest tinygo, so we'll take a look at this and get back to you asap.

File: internal/models/req_meta_data.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package models
```
Comment (wikiwong) [inline internal/models/req_meta_data.go]: I think we can avoid creating this package by defining the `MetaData` struct in the `pdk` package (extism_pdk.go) and calling it `HTTPRequestMeta`.

**Review by wikiwong — CHANGES_REQUESTED** [review]
Thanks for this @syke99, looks great! Just a minor change requested (in addition to merge conflicts)

File: internal/models/req_meta_data.go (hunk 1 above)
Comment (syke99) [reply internal/models/req_meta_data.go]: resolved!! 😁

File: extism_pdk.go [hunk 2]
```diff
@@ -184,11 +183,15 @@ func RemoveVar(key string) {
 	extism_var_set(mem.offset, 0)
 }
 
+type HTTPRequestMeta struct {
+	Url     string            `json:"url"`
+	Method  string            `json:"method"`
+	Headers map[string]string `json:"header"`
```
Comment (nilslice) [inline extism_pdk.go]: technically we support this json field name in the Rust runtime, but for consistency and long-term support I think we should change the JSON struct tag to `headers` here.

**Review by nilslice — CHANGES_REQUESTED** [review]

File: extism_pdk.go (hunk 2 above)
Comment (syke99) [reply extism_pdk.go]: whoops, yeah. was a copy paste error lol. one sec

Comment (syke99) [reply extism_pdk.go]: there we go 😋

**Review by wikiwong — APPROVED** [review]

**Review by nilslice — APPROVED** [review]
LGTM!

**Comment by wikiwong** [comment]
Great update, thanks @syke99 !

**Pull request merged**